src/Chainweb/Pact/PactService.hs

```diff
@@ -622,6 +622,17 @@ withCheckpointer target caller act = mask $ \restore -> do
         finalizeCheckpointer (flip _cpSave $ _blockHash header)
         psStateValidated .= Just header
 
+-- | 'withCheckpointer' but using the cached parent header for target.
+withCurrentCheckpointer
+    :: PayloadCasLookup cas
+    => String
+    -> (PactDbEnv' -> PactServiceM cas (WithCheckpointerResult a))
+    -> PactServiceM cas a
+withCurrentCheckpointer caller act = do
+    ph <- _parentHeader <$> use psParentHeader
+    let target = Just (succ $ _blockHeight ph, _blockHash ph)
```

Comment:
It's probably beyond the scope of this PR, but I find this parameter highly confusing, because the types don't indicate that the block height is't the height of the hash.